tracer/src/Datadog.Trace/Tagging/RabbitMQTags.cs

```diff
@@ -50,5 +51,46 @@ public RabbitMQTags(string spanKind)
 
         [Tag(Trace.Tags.AmqpQueue)]
         public string Queue { get; set; }
+
+        [Tag(Trace.Tags.OutHost)]
+        public string OutHost { get; set; }
+    }
+
+    internal partial class RabbitMQV1Tags : RabbitMQTags
+    {
+        private string _peerServiceOverride = null;
+
+        public RabbitMQV1Tags()
+            : base()
+        {
+        }
+
+        public RabbitMQV1Tags(string spanKind)
+            : base(spanKind)
+        {
+        }
+
+        // Use a private setter for setting the "peer.service" tag so we avoid
+        // accidentally setting the value ourselves and instead calculate the
+        // value from predefined precursor attributes.
+        // However, this can still be set from ITags.SetTag so the user can
+        // customize the value if they wish.
+        [Tag(Trace.Tags.PeerService)]
+        public string PeerService
+        {
+            get => _peerServiceOverride ?? OutHost;
+            private set => _peerServiceOverride = value;
+        }
+
+        [Tag(Trace.Tags.PeerServiceSource)]
+        public string PeerServiceSource
+        {
+            get
+            {
+                return _peerServiceOverride is not null
+                           ? Tags.PeerService
+                           : Tags.OutHost;
```

Comment:
Our metadata rules are complaining that the inbound spans have an extra tag `"_dd.peer.service.source"="out.host"`. I think you might want to do something like this (not sure if our style rules will like this though):

```suggestion
                return _peerServiceOverride is not null
                           ? Tags.PeerService
                           : OutHost is not null
                               ? Tags.OutHost
                               : null;
```

Reply:
After this fix, I think our snapshot comparisons will start to fail, in which case they'll need to be updated to use "out.host" instead of "server.address"

Reply:
Can you have a look at what I ended up doing? Copying it here

```csharp
[Tag(Trace.Tags.PeerServiceSource)]
public string PeerServiceSource
{
    get
    {
        return _peerServiceOverride is not null
                   ? Tags.PeerService
                   : SpanKind.Equals(SpanKinds.Client) || SpanKind.Equals(SpanKinds.Producer)
                       ? Tags.OutHost
                       : null;
    }
}
```

Your suggestion would work because inbound spans don't have the `out.host` tag. Did this for two reasons:
* Being more explicit and intentional, the logic shows exactly why you want the tag to be null
* Consistency with other integrations e.g. kafka: `bootstrap.servers` is always present even for inbound flows, but we still don't want to show `_dd.peer.service.source`